apm-lambda-extension/logsapi/subscribe_test.go

```diff
@@ -0,0 +1,148 @@
+	expectedTypes := []EventType{Platform}
+	expectedBufferingCfg := BufferingCfg{
+		MaxItems:  10000,
+		MaxBytes:  262144,
+		TimeoutMS: 25,
+	}
+
+	// Create aws runtime API server and handler
+	awsRuntimeApiServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		request_bytes, _ := ioutil.ReadAll(r.Body)
+		req := SubscribeRequest{}
+		json.Unmarshal(request_bytes, &req)
+		// Validate the subscription request
+		assert.Equal(t, req.BufferingCfg, expectedBufferingCfg)
+		assert.Equal(t, req.EventTypes, expectedTypes)
+	}))
+	defer awsRuntimeApiServer.Close()
+
+	// Set the Runtime server address as an env variable
+	os.Setenv("AWS_LAMBDA_RUNTIME_API", awsRuntimeApiServer.Listener.Addr().String())
+
+	// Subscribe to the logs api and start the http server listening for events
+	err := Subscribe(ctx, "testID", []EventType{Platform}, out)
+	if err != nil {
+		t.Logf("Error subscribing, %v", err)
+		t.Fail()
+		return
+	}
+	defer logsAPIServer.Close()
+
+	// Create a request to send to the logs listener
+	platformDoneEvent := `{
+		"time": "2021-02-04T20:00:05.123Z",
+		"type": "platform.runtimeDone",
+		"record": {
+		   "requestId":"6f7f0961f83442118a7af6fe80b88",
+		   "status": "success"
+		}
+	}`
+	body := []byte(`[` + platformDoneEvent + `]`)
+	url := "http://" + logsAPIListener.Addr().String()
+	req, err := http.NewRequest("GET", url, bytes.NewReader(body))
+	if err != nil {
+		t.Log("Could not create request")
+	}
+
+	// Send the request to the logs listener
+	client := &http.Client{}
+	go func() {
+		_, err = client.Do(req)
+	}()
+
+	if err != nil {
+		t.Logf("Error fetching %s, [%v]", url, err)
+		t.Fail()
+	} else {
+		event := <-out
+		assert.Equal(t, event.Record.RequestId, "6f7f0961f83442118a7af6fe80b88")
+	}
+}
+
+func TestSubscribeWithBadLogsRequest(t *testing.T) {
+	listenerHost = "localhost"
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	out := make(chan LogEvent)
+
+	// Create aws runtime API server and handler
+	awsRuntimeApiServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
+	defer awsRuntimeApiServer.Close()
+
+	// Set the Runtime server address as an env variable
+	os.Setenv("AWS_LAMBDA_RUNTIME_API", awsRuntimeApiServer.Listener.Addr().String())
+
+	// Subscribe to the logs api and start the http server listening for events
+	err := Subscribe(ctx, "testID", []EventType{Platform}, out)
+	if err != nil {
+		t.Logf("Error subscribing, %v", err)
+		t.Fail()
+		return
+	}
+	defer logsAPIServer.Close()
+
+	// Create a request to send to the logs listener
+	logEvent := `{"type": "invalid"}`
+	body := []byte(`[` + logEvent + `]`)
+	url := "http://" + logsAPIListener.Addr().String()
+	req, err := http.NewRequest("GET", url, bytes.NewReader(body))
+	if err != nil {
+		t.Log("Could not create request")
+	}
+
+	// Send the request to the logs listener
+	client := &http.Client{}
+	resp, _ := client.Do(req)
```

Comment:
`http.DefaultClient`. And it's always a good idea to `require.NoError(t, err)` when you have a method/func that returns an error